sdk/communication/Azure.Communication.Common/src/CommunicationIdentifier.cs

```diff
@@ -27,6 +27,31 @@ public abstract class CommunicationIdentifier : IEquatable<CommunicationIdentifi
         public override bool Equals(object obj)
             => obj is CommunicationIdentifier other && Equals(other);
 
+        /// <summary>
+        /// Overrides the equality operator.
+        /// </summary>
+        /// <param name="left">The first identifier to compare.</param>
+        /// <param name="right">The second identifier to compare.</param>
+        /// <returns>True if the types and <see cref="RawId"/> match.</returns>
+        public static bool operator ==(CommunicationIdentifier left, CommunicationIdentifier right)
+        {
+            return Equals(left, right) && left.GetType() == right.GetType();
+        }
+
+        /// <summary>
+        /// Overrides the non-equality operator.
+        /// </summary>
+        /// <param name="left">The first identifier to compare.</param>
+        /// <param name="right">The second identifier to compare.</param>
+        /// <returns>True if the types or <see cref="RawId"/> values are different.</returns>
+        public static bool operator !=(CommunicationIdentifier left, CommunicationIdentifier right)
+        {
+            return !Equals(left, right) || left.GetType() != right.GetType();
+        }
+
+        /// <summary> Converts a string to a <see cref="CommunicationIdentifier"/>. </summary>
+        public static implicit operator CommunicationIdentifier(string value) => FromRawId(value);
```

Comment:
Implicit cast operators can be quite surprising, do we want that you can pass any string to methods that take a CommunicationIdentifier? I don't know if there was any feedback that prompted you to add this conversion.

I would be less concerned if this was an `explicit` cast operator, and the C# docs also say that conversions that can throw shouldn't be implicit: https://docs.microsoft.com/en-us/dotnet/csharp/language-reference/operators/user-defined-conversion-operators

Reply:
Yes, there was one specific thing that prompted me to add the implicit operator and that was the `CommunicationCloudEnvironment` which also overrides the equality operators. 

https://github.com/Azure/azure-sdk-for-net/blob/aed1cf1cc8f335afd393ca73cec6b0627ce5d382/sdk/communication/Azure.Communication.Common/src/CommunicationCloudEnvironment.cs#L34

I agree that we should stick to the guidelines and use an `explicit` operator. 

Pity that changing it for the `CommunicationCloudEnvironment` would be a breaking change...as it can throw as well.